pkg/sign/notary.go

```diff
@@ -187,63 +233,63 @@ func (ns NotarySigner) Sign(images []string) error {
 	defer w.Stop()
 	for retries > 0 {
 		fmt.Printf("Trying to sign %s. %v retries remaining...\n", sImg, retries)
-		// wait for ticker to run
 		<-w.C
-		resp, err := ns.c.Do(req)
+
+		// Send the HTTP request
+		resp, err := client.Do(req)
 		if err != nil {
-			return fmt.Errorf("notary request: %w", err)
+			return fmt.Errorf("failed to send request: %w", err)
 		}
-		status = resp.Status
+		defer resp.Body.Close()
+
+		// Read the response body
 		respMsg, err = io.ReadAll(resp.Body)
 		if err != nil {
-			return fmt.Errorf("body read: %w", err)
+			return fmt.Errorf("failed to read response: %w", err)
 		}
 		switch resp.StatusCode {
-		case http.StatusOK:
-			// response was fine. Do not need anything else
+		case http.StatusAccepted:
 			fmt.Printf("Successfully signed images %s!\n", sImg)
 			return nil
 		case http.StatusUnauthorized, http.StatusForbidden, http.StatusBadRequest, http.StatusUnsupportedMediaType:
 			return ErrBadResponse{status: status, message: string(respMsg)}
 		}
 		retries--
 	}
-	fmt.Println("Reached all retries. Stopping.")
-	return ErrBadResponse{status: status, message: string(respMsg)}
+
+	return fmt.Errorf("failed to sign images: %s", string(respMsg))
```

Comment:
Why do not return wrapped ErrBadResponse error. It contains `respMsg` and `status`. This way we will have more information aboutfailure.
```suggestion
	return fmt.Errorf("failed to sign images: %w", ErrBadResponse{status: status, message: string(respMsg)})
```

Reply:
Done